lib/private/Files/Storage/DAV.php

```diff
@@ -587,7 +587,7 @@ public function stat($path) {
 				return false;
 			}
 			return [
-				'mtime' => strtotime($response['{DAV:}getlastmodified']),
+				'mtime' => isset($response['{DAV:}getlastmodified']) ? strtotime($response['{DAV:}getlastmodified']) : false,
 				'size' => (int)isset($response['{DAV:}getcontentlength']) ? $response['{DAV:}getcontentlength'] : 0,
```

Comment:
Since the value of `$response['{DAV:}getcontentlength']` is used without modification if it exists, we could write
```suggestion
				'size' => $response['{DAV:}getcontentlength'] ?? 0,
```
Just a suggestion as I was looking at this line again.

Reply:
Not sure which version of php introduced it. We need to support v7.3 for NC24 I think.

Reply:
> We need to support v7.3 for NC24 I think

Far too late https://github.com/nextcloud/server/pull/29286 :)

Reply:
The null coalescing operator was [introduced with php 7](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op). ([php rfc](https://wiki.php.net/rfc/isset_ternary))

So as long as we keep the `(int)` cast because `getcontentlength` is represented as string (I think?) that should be fine. However, I don't have a strong opinion about it. If you want to keep it the way it is, that's fine, too :)

Reply:
Actually, `(int)null` give `0`, so it would probably work to keep the following:

```suggestion
 			'size' => (int)$response['{DAV:}getcontentlength']),
```

But it generates a warning `PHP Notice:  Undefined index: ko in php shell code on line 1`. Same with `??`. So let's not change the current way :).

Reply:
Ok. I'm wondering how you get this notice? Executing the following snippet with php 7.4.3 results in a notice, which (I think) is the same as yours:

```php
<?php

$arr = ['length' => '123'];

// key defined in arr
$length = (int)$arr['length'] ?? 0;
$length2 = (int)($arr['length'] ?? 0);

// key not defined in arr
$not_def = (int)$arr['not_defined'] ?? 0; // notice here
$not_def2 = (int)($arr['not_defined'] ?? 0);


echo gettype($length) . ": " .$length ."\n";
echo gettype($length2) . ": " .$length2 ."\n";
echo gettype($not_def) . ": " .$not_def ."\n";
echo gettype($not_def2) . ": " .$not_def2 ."\n";
```

`PHP Notice:  Undefined index: not_defined in /home/user/nullthing.php on line 10`
The statement one line 11 does basically the same but with extra brackets around null coalescing operator. So I guess they are important. Did you use them? In your suggested change, I see one closing round bracket after the square bracket at the end of the line but no opening bracket in front of `$response`. So, I'm not sure if you forgot to remove the closing bracket or to add the opening bracket :'D

Reply:
Forget the parenthesis, thanks for the example :) I am fine with your change

Reply:
Sounds good. I applied the suggestion